gen3git.py

```diff
@@ -329,9 +337,17 @@ def main(args=None):
             # account for case where no start tag and ver is same as upper_bound
             if upper_bound and ver == upper_bound:
                 continue
+            # find the most recent tag that's lower than `stop_tag`
+            start_ver = parse_version(start_tag.name) if start_tag else None
             if (
                 not start_tag
-                or ver > parse_version(start_tag.name)
+                or (
+                    # ignore non-semantic tags
+                    isinstance(ver, Version)
+                    and isinstance(start_ver, Version)
+                    # check if this tag is more recent than the current `start_tag`
+                    and ver > parse_version(start_tag.name)
```

Comment:
```suggestion
                    and ver > start_ver
```
saves one func call?

Reply:
oops i forgot to update it, thanks